Fixes for steered MD tutorial and update to RBFE tutorial

mb2055 opened the pull request:
This brings the tutorials in line with the latest version of BioSimSpace. 
Changes are in two places:
1) 04_FEP/02_RBFE - Now able to use the latest version of Cinnabar (also retains support for older versions). @annamherz  I wasn't sure if there was a specific reason why cinnabar 0.3.0 was the one used, so I made sure the notebook still supports this version, but also later versions.
2) 03_steered_md - There had been some changes to the API and the tutorial was out of date. @lohedges I wasn't completely sure about the best choices for `align_selection` and  `rmsd_selection` so I chose the same region for both, if there's a better choice here its easy to change.

Commit:
Updates RBFE notebook to support cinnabar 0.4.1

Commit:
Update README to new cinnabar version

Commit:
Updates CollectiveVariable.RMSD to new API

annamherz commented:
> 1. 4_FEP/02_RBFE - Now able to use the latest version of Cinnabar (also retains support for older versions). @annamherz  I wasn't sure if there was a specific reason why cinnabar 0.3.0 was the one used, so I made sure the notebook still supports this version, but also later versions.

Hi! From what I recall, there wasn't a specific reason, mainly that the analysis was stable with that version at the time of writing :smile:

lohedges commented:
Just to note that the updated steered molecular dynamics tutorial requires an update to an input file within the data archive. Since these aren't versioned, I suggest creating a new archive on SharePoint, e..g `data2.tar.bz2`, then downloading it to the _same_ archive name within the download script, i.e. `data.tar.bz2`. That way we don't break backwards compatibility for old versions of the tutorial.

@jmichel80: [This](https://github.com/user-attachments/files/18941585/reference.pdb.gz) file needs to replace `reference.pdb` in the `data.tar.bz2` archive. The SharePoint link appears to be from your account, so it would probably be easiest if edit this. I suggest doing:

1) Download and upack the archive.
2) Replace the `reference.pdb` file with the one posted here.
3) Create a new archive call `data2.tar.bz2` (or give it a more meaningful name if you want to version it according to year, etc) and upload.
4) Create a download link for the new archive and update in the `get_tutorials.py` file. (I can do this if needed. You'd just need to share the link.)

Commit:
Updates  to be consistent with older versions.

Commit:
Adds link to snakemake workflow to RBFE setup tutorial

Commit:
Updates get_tutorial links to new location, and alters the steered md reference.pdb

Also adds reference to the snakemake workflow in the LIVECOMS article and RBFE notebook

mb2055 commented:
Now up to date, with some new references to the RBFE snakemake workflow. Also updated `reference.pdb` to the new version, and updated links within all `get_tutorial` scripts with the new location.

lohedges commented:
Thanks. We'll still need to update the README to use Python 3.11 by default along with the latest BioSImSpace. I can try testing this locally next week. (Or in a container.)

Commit:
Post-testing changes, including slight update to node filepaths and updated README

Commit:
Blacken node running notebook

Commit:
Update python version in README

mb2055 commented:
I've now tested the full suite with the latest version of BSS and everything is working now that the node functionality is fixed. 
The only outstanding issue is that of `plumed` - the conda package hasn't been updated for a long time and as such it doesn't work with any version of BSS after 2024.3. I got around this locally by building `plumed` from source and linking the relevant files in PATH, but this isn't really a practical long-term solution. 
As a result of the `plumed` issue the only part of the repo that hasn't been updated is the install instructions, specifically the line `mamba install -n bsstutorials -c openbiosim -c conda-forge biosimspace=2023.5.0 gromacs=2023.1 ambertools=23.3 plumed=2.9.0 cinnabar=0.3.0 pymbar=3 alchemlyb=1.0.1` as updating this to the version I used for testing would just result in an environment that doesn't resolve.

mb2055 commented:
Just adding a note here that following the BSS refactor, which switches to using the new sire API, there are a couple of changes that are needed to the `01_introduction` tutorials:
- 01_introduction.ipynb  has multiple instances of `propertyKeys` that need to be changed to `property_keys`
- 02_molecular_setup.ipynb  once again has multiple instances of `propertyKeys` that need to be changed to `property_keys`
